### PR TITLE
docs: Move development documentation into `docs/development/`

### PR DESCRIPTION
**Pull request by peter-b**
This patch moves almost all documentation related to development into
the `docs/` subdirectory.  The only exceptions are:
- the toplevel `CONTRIBUTING.md` file, which needs to be located there
  since GitHub looks for it to link from the PR creation page
- the documentation contributions guide, which is linked directly from
  a bunch of places; it might be worth reconsidering its location at
  some point in the future

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -75,13 +75,13 @@ On Linux or Mac, you can quickly build LiveCode by installing basic development
...
-| Target platform                             | Host platforms    |
-| ------------------------------------------- | ----------------- |
-| [mac, ios](INSTALL-mac.md)                  | mac               |
-| [win](INSTALL-win.md)                       | win, linux (Wine) |
-| [linux](INSTALL-linux.md)                   | linux             |
-| [android](INSTALL-android.md)               | mac, linux        |
-| [emscripten (html5)](INSTALL-emscripten.md) | linux             |
+| Target platform                                            | Host platforms    |
+| ---------------------------------------------------------- | ----------------- |
+| [mac, ios](docs/development/build--mac.md)                 | mac               |
```
Comment (livecodeali) [inline README.md]: `build-mac.md`

**Comment by livecodeali** [comment]
@livecode-vulcan review ok 4e226df

**Comment by livecode-vulcan** [comment]
💙 review by @livecodeali ok 4e226df60cbd7bec19b1c0a60eff7c7a68d5e8d2

**Comment by livecode-vulcan** [comment]
😎 test success 4e226df60cbd7bec19b1c0a60eff7c7a68d5e8d2
- try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/447)
- try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/452)
- try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/447)
- try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/447)
- try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/453)
- try-community-win-x86: [success](https://vulcan.livecode.com/builders/try-community-win-x86/builds/447)
- try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/447)

**Pull request merged**